src/components/PopupWithForm.js

```diff
@@ -20,8 +21,14 @@ export default class PopupWithForm extends Popup {
 
   reset() {
     this._form.reset();
-    // the review comment that was sent got lost and would not popup for this section so i couldnt see what needed to be fixed
-    //i will remove these comments when it gets checked again i had to try to remember what was said by memory
+  }
+
+  renderLoading(loading) {
+    if (loading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
+    }
```

Comment:
Needs correction (line 39): Please don't reset the form right after calling the submit handler. You should do that from index.js only after the successful api response, so the form won't be cleared if an error occurred